core/src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,10 +1,27 @@
 package pureconfig
 
+/**
+ * A mapping between case class fields and their respective keys in the config.
+ */
 trait ConfigFieldMapping extends (String => String) {
+
   def apply(fieldName: String): String
+
+  /**
+   * Returns a `ConfigFieldMapping` that uses this mapping with some exceptions.
+   *
+   * @param exceptions the exceptions for this mapping as pairs (field, configKey)
```

Comment:
@ruippeixotog I agree that I was wrong when I said "globally throughout the configuration." I didn't realize the scope that the `ConfigFieldMapping` was applied within was a `ProductHint`. (I probably should, but I didn't.)

Probably nothing needs to be added. I'm skeptical I could inform without confusing.

Sorry.